### PR TITLE
TMDM-11567 Querying data records having null/empty values

### PR DESCRIPTION
**Pull request by zfkang-talend**
**What is the current behavior?** (You should also link to an open issue here)
In MQL, **isEmpty** is not pure, it parsed to **isEmpty or isNull** when parse to condition.

**What is the new behavior?**
Changed isEmpty to pure, only isEmpty when parsing to condition.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by zfkang-talend** [comment]
Build Jenkins success, have no failed Junit case. 
Linke: http://192.168.31.210:8080/view/Master/job/Master_03_TMDMEE_FOR-JIRA_Master/45/

File: org.talend.mdm.query/pom.xml [hunk 1]
```diff
@@ -50,4 +50,43 @@
         </dependency>
     </dependencies>
 
+    <profiles>
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.query/pom.xml]: If without this new added part, does the previous build also work?

File: org.talend.mdm.query/src/com/amalto/core/query/user/UserQueryBuilder.java [hunk 2]
```diff
@@ -599,10 +599,11 @@ public static Condition emptyOrNull(TypedExpression field) {
             if (!fieldMetadata.isMany() && fieldMetadata instanceof ReferenceFieldMetadata) {
                 return new IsNull(new Field(fieldMetadata));
             }
-        }                
+        }
         // Only do a isEmpty operator if field type is string, for all other known cases, isNull is enough.
         if (Types.STRING.equals(field.getTypeName())) {
-            return new BinaryLogicOperator(isEmpty(field), Predicate.OR, isNull(field));
+            Condition condition = or(new IsEmpty(field), isNull(field));
+            return new BinaryLogicOperator(condition, Predicate.OR, isNull(field));
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.query/src/com/amalto/core/query/user/UserQueryBuilder.java]: From above two lines, `isNull` was used in two places, is this expected?

File: org.talend.mdm.query/pom.xml (hunk 1 above)
Comment (zfkang-talend) [reply org.talend.mdm.query/pom.xml]: yes, but if you build project org.talend.mdm.query in the local computer, you should manual to copy the jar file to tomcat path folder.

File: org.talend.mdm.query/src/com/amalto/core/query/user/UserQueryBuilder.java (hunk 2 above)
Comment (VictorLi-talendbj) [reply org.talend.mdm.query/src/com/amalto/core/query/user/UserQueryBuilder.java]: Do we have Junit case for this change?

**Pull request merged**